ReSharper.FSharp/src/FSharp.ExternalFormatter.Protocol/src/ExternalFormatterProcess.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using JetBrains.Application.Processes;
+using JetBrains.Application.Threading;
+using JetBrains.Core;
+using JetBrains.DataFlow;
+using JetBrains.Diagnostics;
+using JetBrains.Lifetimes;
+using JetBrains.Platform.RdFramework.ExternalProcess;
+using JetBrains.Rd;
+using JetBrains.Rider.FSharp.ExternalFormatter.Server;
+using JetBrains.Util;
+
+namespace FSharp.ExternalFormatter.Protocol
+{
+  public class
+    ExternalFormatterProcess : ProtocolExternalProcess<RdFSharpExternalFormatterModel, ExternalFormatterConnection>
+  {
+    protected override string Name => "External Formatter";
+
+    protected override RdFSharpExternalFormatterModel CreateModel(Lifetime lifetime, IProtocol protocol) =>
+      new RdFSharpExternalFormatterModel(lifetime, protocol);
+
+    protected override ExternalFormatterConnection CreateConnection(Lifetime lifetime,
+      RdFSharpExternalFormatterModel model, IProtocol protocol, StartupOutputWriter outputWriter, int processId,
+      Signal<int> processUnexpectedExited) =>
+      new ExternalFormatterConnection(lifetime, model, protocol, outputWriter, processId,
+        processUnexpectedExited);
+
+    protected override ProcessStartInfo GetProcessStartInfo(int port)
+    {
+      var launchPath = GetType().Assembly.GetPath().Directory.Combine(ProtocolConstants.PROCESS_FILENAME);
+      Assertion.Assert(launchPath.ExistsFile, $"can't find '{ProtocolConstants.PROCESS_FILENAME}'");
+
+      return new ProcessStartInfo
+      {
+        Arguments =
+          $"{port} \"{ProtocolConstants.LogFolder.Combine($"{DateTime.UtcNow:yyyy_MM_dd_HH_mm_ss_ffff}.log")}\"",
```

Comment:
Should it mention something about Fantomas host or F#? Looks like a common class at a glance.

Reply:
pid of the parent process is passed in [environment variables](https://github.com/JetBrains/resharper-fsharp/pull/252/files/92d10c62dd2f875243d5294878ea0643248e6c77#diff-2e386d439455966e46f2935fbdff0b0df204fe948b9fd71427d3916b71e6a446R54)

Reply:
?

Reply:
I've tried to clarify my comment.

Reply:
nobody in the platform did this because the namespace already contains the necessary information. But we can (: